src/library/VUMeter.js

```diff
@@ -0,0 +1,93 @@
+ *     If the decible is lower than this value, it cosider as slient.
+ * @param {AnalyserNode} analyserNode The analysis node
+ *     which connect with the audio context.
+ */
+class VUMeter {
+  constructor(canvasId, minDecible, analyserNode) {
+    this.canvas_ = document.querySelector(canvasId);
+    this.canvasContext_ = this.canvas_.getContext('2d');
+    this.width_ = this.canvas_.width;
+    this.height_ = this.canvas_.height;
+    this.currentX_ = 0;
+    this.previousY_ = this.height / 2;
+    // Due to the calculation and graph convinience,
+    // we set it to positive
+    this.minDisplayDecible_ = Math.abs(minDecible);
+    this.fifoArray_ = new FIFOArray(8);
+    this.analyser_ = analyserNode;
+    // This is the minimum fftSize which we are able to have.
+    // By using size 32, we can collect the most accurate data.
+    this.analyser_.fftSize = 32;
+    this.bufferLength_ = analyserNode.frequencyBinCount;
+    this.dataArray_ = new Uint8Array(this.bufferLength_);
+  }
+
+  /**
+   * This is the create function for VUMeter
+   */
+  draw() {
+    this.analyser_.getByteTimeDomainData(this.dataArray_);
+    
+    let formatDataArray = Array(this.dataArray_.length);
+    for(let i = 0, length = this.dataArray_.length; i < length; i++){
+        formatDataArray[i] = this.dataArray_[i] / 128.0 - 1.0;
+    }
+    
+    const rootMeanSquare = this.rms(formatDataArray);
+    const decibel = 20 * Math.log10(rootMeanSquare);
+    let absDecibel = Math.abs(decibel);
+    this.fifoArray_.add(absDecibel);
+    let minDecibel = this.fifoArray_.getMinValue();
+    let meterHeight = minDecibel > this.minDisplayDecible_ ?
+        0 : minDecibel / this.minDisplayDecible_ * this.height_;
+
+    this.canvasContext_.clearRect(0, 0, this.width_, this.height_);
+
+    this.canvasContext_.fillStyle = '#000';
+    this.canvasContext_.fillRect(0, 0, this.width_, this.height_);
+    this.canvasContext_.fillStyle = 'green';
+    this.canvasContext_.fillRect(0, 0, this.width_, meterHeight);
+
+    this.canvasContext_.strokeStyle = 'black';
+    this.canvasContext_.lineWidth = 1;
+    this.canvasContext_.globalAlpha = 0.3;
+    this.canvasContext_.beginPath();
+    this.canvasContext_.moveTo(0, this.height_ * 0.2);
+    this.canvasContext_.lineTo(this.width_, this.height_ * 0.2);
+    this.canvasContext_.moveTo(0, this.height * 0.4);
+    this.canvasContext_.lineTo(this.width_, this.height_ * 0.4);
+    this.canvasContext_.moveTo(0, this.height_ * 0.6);
+    this.canvasContext_.lineTo(this.width_, this.height_ * 0.6);
+    this.canvasContext_.moveTo(0, this.height_ * 0.8);
+    this.canvasContext_.lineTo(this.width_, this.height_ * 0.8);
+    this.canvasContext_.stroke();
+  }
+
+  /**
+   * Calculation of the root mean square of input array
+   * @param {array} inputArray array to be calculated
+   * @returns {number} The root mean square of array
+   */
+  rms(inputArray) {
+    return Math.sqrt(
+      inputArray
+          .map( val => (val * val))
```

Comment:
This creates a new array: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

I would advise against using this in the active draw function.